common/flatpak-context.c

```diff
@@ -852,11 +916,60 @@ flatpak_context_parse_filesystem (const char             *filesystem_and_mode,
   return FALSE;
 }
 
+/* Note: This only works with valid keys, i.e. they passed flatpak_context_parse_filesystem */
+static gboolean
+flatpak_filesystem_key_in_home (const char *filesystem)
+{
+  /* "home" is definitely in home */
+  if (strcmp (filesystem, "home") == 0)
+    return TRUE;
+
+  /* All the other special fs:es are non-home.
+   * Note: This considers absolute paths that are in the homedir as non-home.
+   */
+  if (g_strv_contains (flatpak_context_special_filesystems, filesystem) ||
+      g_str_has_prefix (filesystem, "/"))
+    return FALSE;
+
+  /* Files in xdg-run are not in home */
+  if (g_str_has_prefix (filesystem, "xdg-run"))
+    return FALSE;
+
+  /* All remaining keys (~/, xdg-data, etc) are considered in home,
+   * Note: technically $XDG_HOME_DATA could point outside the homedir, but we ignore that.
+   */
+  return TRUE;
+}
+
 static void
 flatpak_context_take_filesystem (FlatpakContext        *context,
                                  char                  *fs,
-                                 FlatpakFilesystemMode  mode)
+                                 FlatpakFilesystemMode  mode,
+                                 FlatpakFilesystemFlags flags)
 {
+  if (mode > FLATPAK_FILESYSTEM_MODE_NONE)
+    {
+      /* --filesystem=foo always clears the RESET flag. */
+      g_assert ((flags & FLATPAK_FILESYSTEM_FLAGS_RESET) == 0);
+
+      if (strcmp (fs, "host") == 0)
+        context->fs_host_reset = FALSE;
```

Comment:
Given the other comment below, and the possibility of having home:rw while at the same time fs_home_reset is TRUE, should we really always reset it to FALSE here?

Like, if the commandline has --nofilesystem=home:reset --filesystem=host, maybe that should be clear all parent home specific things, but  add "home".

Reply:
For example, it may be interesting to do --nofilesystem=home:reset --filesystem=host:ro